### DISCUSSION TITLE
converting humming in the I2S microphone to the correct midi stream

### DISCUSSION THREAD

**Discussion by wimverlindensa** [pschatzmann/arduino-audio-tools#165] [Q&A] [answered]
I was wondering if it is possible to sing ( hum ) in the I2S microphone and have this converted in real time to a midi stream with the correct ( or adjusted to the nearest note ) pitch and amplitude....  

I see only one example mentioning FFT but it says // WARNNG: This sketch is too big to fit on an ESP32 !

Does the framework offers solutions for FFT? ( ESP32 or RP )

Thanks!

**Comment by pschatzmann** [comment]
My gut feeling is that it should be possible using FFT.  Currently this library does not provide any automatic integration, but this should be quite easy to implement.  

Maybe someone else might recommend a library ?  (There are plenty to choose from...)

You can use the VolumePrint class to determine the volume.
You can use the MusicalNotes class to determine the note from the frequency

**Comment by pschatzmann** [comment, marked as answer]
I have added some [new simple FFT functionality](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_audio_f_f_t.html). 

An [example can be found here](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/streams-audiokit-fft).

I also added an entry to the [Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/FFT)

ps. I just added some addtioinal methods to the [MusiclNotes](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_musical_notes.html) to determine the midi note from the frequency...

**Reply by wimverlindensa** [reply]
Thank you! Will test it!
